src/core/public/chrome/ui/header/header.tsx

```diff
@@ -87,6 +87,7 @@ export function Header({
   const isVisible = useObservable(observables.isVisible$, false);
   const isLocked = useObservable(observables.isLocked$, false);
   const [isNavOpen, setIsNavOpen] = useState(false);
+  const [navId] = useState(htmlIdGenerator()());
```

Comment:
NIT: I think we should probably use `useMemo` instead of `useState` here?

Reply:
React reserves the right to forget memoized values, https://reactjs.org/docs/hooks-reference.html#usememo

> You may rely on useMemo as a performance optimization, not as a semantic guarantee. In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without useMemo — and then add it to optimize performance.

However, a ref _could_ be a better fit as it best represents the intent. That said, modifying the `useState` here to use an initialization function is the best performance as it will avoid calling the uuid package in subsequent runs. However, I can call `htmlIdGenerator()()` 10,000 times in 20ms on my MBP, so I'd avoid the performance argument in favor of any readability.

Reply:
> You may rely on useMemo as a performance optimization, not as a semantic guarantee

Thanks, TIL.